vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExhibitionManager.java

```diff
@@ -66,4 +77,50 @@ public IExhibition getStartExhibition() {
         }
         return null;
     }
+
+    /**
+     * Updates the Exhibition with given list of languages. It fetches the language from exhibitionLanguageConfig using code.
+     *  
+     * @param exhibition
+     * @param defaultLanguage 
+     * @param languages
+     */
+    @Override
+    public void updateExhibitionLanguages(Exhibition exhibition, List<String> codes, String defaultLanguage) {
+        List<ExhibitionLanguage> languageMapList = new ArrayList();
+        if(CollectionUtils.isNotEmpty(codes)) {
+            if(defaultLanguage!=null) {
+                codes.add(defaultLanguage);
+            }
+            codes.forEach(code -> {
```

Comment:
I think I would turn this around and iterate over exhibitionLanguageConfig.getExhibitionLanguageList(). You would only need one stream and filter for everything in the codes list. So only one iteration instead of multiple if I'm not mistaken.